lib/plugin.js

```diff
@@ -1,112 +1,45 @@
-function setupBabelHook(babelConfig) {
-  return addHook(
-    (code, filename) => {
-      const { code: compiledCode } = babel.transform(code, {
-        filename,
-        root: APP_ROOT,
-        babelrc: false,
-        ...babelConfig,
-      });
-      compiledFiles.add(filename);
-
-      return `${compiledCode}\nif (exports.default) exports.default.__modulePath = "${filename}"`;
-    },
-    {
-      exts: SUPPORTED_EXTENSIONS.map((ext) => `.${ext}`),
-      ignoreNodeModules: true,
-    }
-  );
-}
-
-function insertComponentsScript(html, assets, assetsPath) {
-  const dom = new JSDOM(html);
-  const { document } = dom.window;
-
-  // This page doesn't have any components to hydrate
-  if (!document.querySelectorAll("[data-hydration-start-id]").length) {
-    return html;
-  }
-
-  const scriptEls = assets.map((src) => {
-    const scriptEl = document.createElement("script");
-    scriptEl.src = `/${assetsPath.replace(/^\.{0,2}\/|\/$/g, "")}/${src}`;
-    return scriptEl;
-  });
-  document.querySelector("body").append(...scriptEls);
-
-  return dom.serialize();
-}
-
-async function insertComponentsScriptForPaths(
-  pagePaths,
-  bundledClientAssets,
-  assetsPath
-) {
-  return Promise.all(
-    pagePaths.map(async (pagePath) => {
-      const absPagePath = path.resolve(process.cwd(), pagePath);
-      const contents = await fs.readFile(absPagePath, "utf8");
-
-      return fs.writeFile(
-        absPagePath,
-        insertComponentsScript(contents, bundledClientAssets, assetsPath)
-      );
-    })
-  );
-}
-
-function removeHelmetDataAttribute(str) {
-  return str.replace(/data-react-helmet="true"/g, "").replace(/ {2,}/g, " ");
-}
-
-async function generateHTML({ bodyContent, helmet }) {
-  let html = dedent`
-    <!doctype html>
-      <html ${removeHelmetDataAttribute(helmet.htmlAttributes.toString())}>
-      <head>
-        ${removeHelmetDataAttribute(helmet.title.toString())}
-        ${removeHelmetDataAttribute(helmet.meta.toString())}
-        ${removeHelmetDataAttribute(helmet.link.toString())}
-      </head>
-      <body ${removeHelmetDataAttribute(helmet.bodyAttributes.toString())}>
-        <div id="content">
-          ${bodyContent}
-        </div>
-      </body>
-    </html>
-  `;
-
-  return html;
-}
+const SUPPORTED_EXTENSIONS = ["js", "jsx", "ts", "tsx"];
 
 module.exports = function eleventyPluginReact(
   eleventyConfig,
-  { babelConfig, assetsPath = "/assets/", postProcess = null } = {}
+  {
+    targets = "last 2 versions, safari >= 12",
+    babelConfig: userBabelConfig = {
+      plugins: [],
+      presets: [],
+    },
+    assetsPath = "/assets/",
+    postProcess = null,
+  } = {}
 ) {
-  if (typeof babelConfig !== "function") {
-    throw new Error("`babelConfig` is required and must be a function.");
-  }
+  const ssrBabelConfig = babelConfig.generate({
+    isBrowser: false,
+    targets,
+    config: userBabelConfig,
+  });
+  const browserBabelConfig = babelConfig.generate({
+    isClientBundle: true,
```

Comment:
`isClientBundle` isn't destructured in `generate()` - should this be `isBrowser`?